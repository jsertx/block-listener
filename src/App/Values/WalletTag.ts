export enum WalletTagName {
	FoundIteratingBlocks = "found-iterating-blocks",
	FoundByIncomingTransfer = "found-by-incoming-transfer",
	AddedManually = "added-manually",
	FoundWithSearcher = "found-with-searcher"
}

export const walletTagNameList = Object.values(WalletTagName);

export interface WalletTag {
	tag: WalletTagName;
	createdAt: Date;
}
