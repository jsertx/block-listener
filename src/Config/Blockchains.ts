export enum BlockchainId {
	Ethereum = "ethereum",
	Polygon = "polygon"
}
