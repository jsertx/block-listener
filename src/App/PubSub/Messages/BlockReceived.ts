import { BlockWithTransactions } from "../../Types/BlockWithTransactions";
import { BlockchainId } from "../../Values/Blockchain";

import { ExecutorMessage } from "../../../Infrastructure/Broker/Executor";
import { Publication } from "../../../Infrastructure/Broker/Rabbitmq/Enums";

export interface BlockReceivedPayload {
	blockchain: BlockchainId;
	block: BlockWithTransactions;
}
export class BlockReceived extends ExecutorMessage<BlockReceivedPayload> {
	constructor(payload: BlockReceivedPayload) {
		super(Publication.BlockReceived, payload);
	}
}
