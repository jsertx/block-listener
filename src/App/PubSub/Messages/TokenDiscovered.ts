import { BlockchainId } from "../../Values/Blockchain";

import { ExecutorMessage } from "../../../Infrastructure/Broker/Executor";
import { Publication } from "../../../Infrastructure/Broker/Rabbitmq/Enums";

export interface TokenDiscoveredPayload {
	blockchain: BlockchainId;
	address: string;
}

export class TokenDiscovered extends ExecutorMessage<TokenDiscoveredPayload> {
	constructor(payload: TokenDiscoveredPayload) {
		super(Publication.TokenDiscovered, payload);
	}
}
