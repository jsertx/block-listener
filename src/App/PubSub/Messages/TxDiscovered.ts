import { BlockchainId } from "../../Values/Blockchain";
import { ethers } from "ethers";
import { ExecutorMessage } from "../../../Infrastructure/Broker/Executor";
import { Publication } from "../../../Infrastructure/Broker/Rabbitmq/Enums";

export interface TxDiscoveredPayload {
	blockchain: BlockchainId;
	hash: string;
	saveUnknown?: boolean;
	txRes?: ethers.providers.TransactionResponse;
	block?: ethers.providers.Block;
}
/**
 * MaxRetries should be a consumer executor as the message does not
 * need to know needs of the consumer
 */
export class TxDiscovered extends ExecutorMessage<TxDiscoveredPayload> {
	constructor(payload: TxDiscoveredPayload) {
		super(Publication.TxDiscovered, payload, {
			maxRetries: 30
		});
	}
}
