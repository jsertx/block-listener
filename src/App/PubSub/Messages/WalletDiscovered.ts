import { BlockchainId } from "../../Values/Blockchain";
import { ExecutorMessage } from "../../../Infrastructure/Broker/Executor";
import { WalletTagName } from "../../Values/WalletTag";
import { AddressRelation } from "../../Entities/Wallet";
import { WalletType } from "../../Values/WalletType";
import { Publication } from "../../../Infrastructure/Broker/Rabbitmq/Enums";

export interface WalletDiscoveredPayload {
	blockchain: BlockchainId;
	alias?: string;
	address: string;
	type: WalletType;
	tags?: WalletTagName[];
	relations?: Omit<AddressRelation, "createdAt">[];
}

export class WalletDiscovered extends ExecutorMessage<WalletDiscoveredPayload> {
	constructor(payload: WalletDiscoveredPayload) {
		super(Publication.WalletDiscovered, payload);
	}
}
