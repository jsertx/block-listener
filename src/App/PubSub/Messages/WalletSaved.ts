import { BlockchainId } from "../../Values/Blockchain";
import { BaseMessage } from "../../../Infrastructure/Broker/BaseMessage";
import { Publication } from "../../../Infrastructure/Broker/Rabbitmq/Enums";

export interface WalletSavedPayload {
	blockchain: BlockchainId;
	address: string;
}

export class WalletSaved extends BaseMessage<WalletSavedPayload> {
	constructor(payload: WalletSavedPayload) {
		super(Publication.WalletSaved, payload);
	}
}
