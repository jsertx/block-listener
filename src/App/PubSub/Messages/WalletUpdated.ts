import { BlockchainId } from "../../Values/Blockchain";
import { BaseMessage } from "../../../Infrastructure/Broker/BaseMessage";
import { Publication } from "../../../Infrastructure/Broker/Rabbitmq/Enums";

export interface WalletUpdatedPayload {
	blockchain: BlockchainId;
	address: string;
}

export class WalletUpdated extends BaseMessage<WalletUpdatedPayload> {
	constructor(payload: WalletUpdatedPayload) {
		super(Publication.WalletUpdated, payload);
	}
}
