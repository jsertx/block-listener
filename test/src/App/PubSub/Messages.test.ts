import { BlockReceived } from "../../../../src/App/PubSub/Messages/BlockReceived";
import { TokenDiscovered } from "../../../../src/App/PubSub/Messages/TokenDiscovered";
import { TxDiscovered } from "../../../../src/App/PubSub/Messages/TxDiscovered";
import { WalletDiscovered } from "../../../../src/App/PubSub/Messages/WalletDiscovered";
import { WalletSaved } from "../../../../src/App/PubSub/Messages/WalletSaved";
import { WalletUpdated } from "../../../../src/App/PubSub/Messages/WalletUpdated";
import { WalletTagName } from "../../../../src/App/Values/WalletTag";
import { BlockchainId } from "../../../../src/Config/Blockchains";
import { BaseMessage } from "../../../../src/Interfaces/IBroker";

type IgnoreTypeOfData = any;
describe("PubSub > Messages", () => {
	const blockReceivedMsg = new BlockReceived({
		block: { block: 1 } as IgnoreTypeOfData,
		blockchain: BlockchainId.Ethereum
	});
	const tokenDiscoveredMsg = new TokenDiscovered({
		address: "0xA",
		blockchain: BlockchainId.Ethereum
	});
	const txDiscoveredMsg = new TxDiscovered({
		blockchain: BlockchainId.Ethereum,
		hash: "0xHASH",
		saveUnknown: true,
		txRes: { txData: "mock" } as IgnoreTypeOfData,
		block: { blockData: "mock" } as IgnoreTypeOfData
	});
	const walletDiscoveredMsg = new WalletDiscovered({
		blockchain: BlockchainId.Ethereum,
		address: "0xAddress",
		tags: [WalletTagName.AddedManually],
		relations: []
	});
	const walletSavedMsg = new WalletSaved({
		address: "0xA",
		blockchain: BlockchainId.Ethereum
	});
	const walletUpdatedMsg = new WalletUpdated({
		address: "0xA",
		blockchain: BlockchainId.Ethereum
	});
	const messages: Array<[string, BaseMessage<any, any>]> = [
		["BlockReceived", blockReceivedMsg],
		["TokenDiscovered", tokenDiscoveredMsg],
		["TxDiscovered", txDiscoveredMsg],
		["WalletDiscovered", walletDiscoveredMsg],
		["WalletSaved", walletSavedMsg],
		["WalletUpdated", walletUpdatedMsg]
	];

	describe.each(messages)("%s", (txt, msg) => {
		it("should build the correct payload", () => {
			expect(msg.payload).toMatchSnapshot();
		});
		it("should get the channel with the blockchain prefix", () => {
			expect(msg.channel).toMatchSnapshot();
		});
	});
});
