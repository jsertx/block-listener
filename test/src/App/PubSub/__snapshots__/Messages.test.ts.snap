// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PubSub > Messages BlockReceived should build the correct payload 1`] = `
Object {
  "maxRetries": 10,
  "payload": Object {
    "block": Object {
      "block": 1,
    },
    "blockchain": "ethereum",
  },
  "processAfter": 1662163200000,
  "retries": 0,
}
`;

exports[`PubSub > Messages BlockReceived should get the channel with the blockchain prefix 1`] = `"block_received"`;

exports[`PubSub > Messages TokenDiscovered should build the correct payload 1`] = `
Object {
  "maxRetries": 10,
  "payload": Object {
    "address": "0xA",
    "blockchain": "ethereum",
  },
  "processAfter": 1662163200000,
  "retries": 0,
}
`;

exports[`PubSub > Messages TokenDiscovered should get the channel with the blockchain prefix 1`] = `"token_discovered"`;

exports[`PubSub > Messages TxDiscovered should build the correct payload 1`] = `
Object {
  "maxRetries": 30,
  "payload": Object {
    "block": Object {
      "blockData": "mock",
    },
    "blockchain": "ethereum",
    "hash": "0xHASH",
    "saveUnknown": true,
    "txRes": Object {
      "txData": "mock",
    },
  },
  "processAfter": 1662163200000,
  "retries": 0,
}
`;

exports[`PubSub > Messages TxDiscovered should get the channel with the blockchain prefix 1`] = `"tx_dicovered"`;

exports[`PubSub > Messages WalletDiscovered should build the correct payload 1`] = `
Object {
  "maxRetries": 10,
  "payload": Object {
    "address": "0xAddress",
    "blockchain": "ethereum",
    "relations": Array [],
    "tags": Array [
      "added-manually",
    ],
    "type": "whale.wallet",
  },
  "processAfter": 1662163200000,
  "retries": 0,
}
`;

exports[`PubSub > Messages WalletDiscovered should get the channel with the blockchain prefix 1`] = `"wallet_discovered"`;

exports[`PubSub > Messages WalletSaved should build the correct payload 1`] = `
Object {
  "address": "0xA",
  "blockchain": "ethereum",
}
`;

exports[`PubSub > Messages WalletSaved should get the channel with the blockchain prefix 1`] = `"wallet_saved"`;

exports[`PubSub > Messages WalletUpdated should build the correct payload 1`] = `
Object {
  "address": "0xA",
  "blockchain": "ethereum",
}
`;

exports[`PubSub > Messages WalletUpdated should get the channel with the blockchain prefix 1`] = `"wallet_updated"`;
